runtime/near-vm-runner/src/vm_kind.rs

```diff
@@ -30,3 +30,13 @@ pub enum VMKind {
     /// NearVM.
     NearVm,
 }
+
+impl VMKind {
+    pub fn normalize(self) -> Self {
```

Comment:
Shouldn't this happen when the vm_kind is first initialized in wasm_config? Otherwise it's easy to forget to call normalize.

Reply:
The issue is that this is initialized from the parameter files, and so is constructed directly through deserialization in most interesting cases. There may be a good way to hook into deserialization code here, but admittedly, I would find it pretty weird if deserializing "wasmer2" produced `VMKind::Wasmtime` too.

Reply:
To be honest I think we should fail if the parameter file specifies a vm kind that is not supported on given architecture. Ideally the vm kind should be inferred directly from the architecture are the user should not be given any choice right? 

Anyway, it's not the worst thing in the world. I'm not going to block you on that so feel free to merge this PR as is.

Reply:
Unfortunately this is not that easy. I mean we could have parameter files that are specific to macOS machines, but two sets of parameter files between mainnet and testnet are already fairly onerous to maintain on their own. And adding an ability to express platform-conditional parameters in the yaml files would also be pretty contentious I imagine.

> Ideally the vm kind should be inferred directly from the architecture are the user should not be given any choice right?

This is how it used to be until very recently. Unfortunately this has caused a fair number of problems (for instance we couldn’t use `--all-features` to compile code) so I ended up moving this configuration over to the runtime-time in https://github.com/near/nearcore/pull/9539. Doing a compile time decision will also become unfeasible once we make further progress on limited replayability…

---

For something that is only relevant in testing and has very clear-cut failure modes, I think a method like this right now is the best compromise.